### PR TITLE
Move benchmark code from nox to bm_runner

### LINKED ISSUE #281: Introduce `bm_runner.py` to iris-esmf-regrid
## 📰 Custom Issue

[See the existing example in Iris](https://github.com/SciTools/iris/blob/main/benchmarks/bm_runner.py).

Motive:

- More flexible - not tied to Nox
- Can introduce more customisation since it is a freeform Python script

### PR DESCRIPTION
**Pull request by stephenworsley**
Closes #281

### CODE REVIEW

**Review by pp-mo — CHANGES_REQUESTED** [review]
I've checked out the function of this now, and I'm happy it does function and contain what we need.
Just a few irrelevant references to fix I think (since you already ditched the unwanted Cperf code)

However, getting to grips with this has made me quite concerned about the future relationship of all this to the Iris benchmarking code, which it is similar to but significantly different.
 - for instance, I'm pleased that we didn't change noxfile.py here, but I'm pretty concerned at how that relates to the Iris code : a lot in common but multiple significant differences
  - on the other hand, the change here to asv_delegated_conda.py just keeps it exactly the same as in Iris, so that is easy to track + could be done with simple "templating" as [recently introduced into scitools/.github](https://github.com/SciTools/.github/pull/23).
The task of generalising so these two implementations can "share and differ as required" seems attractive but is clearly already non-trivial.  So, I'm hoping we can come up with better ways to do this in future, but **_not now_** !

File: benchmarks/bm_runner.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+            hashfile.writelines([merge_base, "\n", head_sha])
+            hashfile.flush()
+            commit_range = f"HASHFILE:{hashfile.name}"
+            asv_command = shlex.split(ASV_HARNESS.format(posargs=commit_range))
+            _subprocess_runner([*asv_command, *args.asv_args], asv=True)
+
+        _asv_compare(merge_base, head_sha)
+
+
+class _CSPerf(_SubParserGenerator, ABC):
```
[the remark this answers is not in the document]
Comment (stephenworsley) [reply benchmarks/bm_runner.py:227]: I was aiming for similarity with the iris structure where possible, though I agree that it doesn't make much sense in tis instance.

**Review by pp-mo — CHANGES_REQUESTED** [review]
Unfortunately I spotted a couple more problems which I think need fixing.
I did check local operation, and it does work :+1:

**Review by pp-mo — APPROVED** [review]

**Comment by pp-mo** [comment]
Great stuff @stephenworsley -- thanks for your patience !

**Pull request merged**